### PR TITLE
[maskedtensor] Overview tutorial [1/4]

### PR DESCRIPTION
**Pull request by george-qi**
Restructured the PR stack from https://github.com/pytorch/tutorials/pull/2042 

The new format will be:

PR #1: Overview Tutorial
- [Old overview](https://github.com/pytorch/tutorials/pull/2042)
- Why is MaskedTensor useful
    - [Distinguish 0 and NaN gradient](https://github.com/pytorch/tutorials/pull/2044)
    - [Safe softmax tutorial](https://github.com/pytorch/tutorials/pull/2045)
    - [Missing nan ops](https://github.com/pytorch/tutorials/pull/2046)

PR #2: [Sparsity](https://github.com/pytorch/tutorials/pull/2043)

PR #3: [Adagrad sparse semantics](https://github.com/pytorch/tutorials/pull/2047), i.e. what does MaskedTensor make easier

PR #4: Advanced semantics, e.g. NumPy differences and reduction semantics from [Old overview](https://github.com/pytorch/tutorials/pull/2042)

### CODE REVIEW

File: prototype_source/maskedtensor_overview.rst [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+and illustrate how :class:`MaskedTensor` can solve these problems.
+
+Distinguishing between 0 and NaN gradient
+-----------------------------------------
+
+One issue that :class:`torch.Tensor` runs into is the inability to distinguish between gradients that are
+undefined (NaN) vs. gradients that are actually 0. Because PyTorch does not have a way of marking a value
+as specified/valid vs. unspecified/invalid, it is forced to rely on NaN or 0 (depending on the use case), leading
+to unreliable semantics since many operations aren't meant to handle NaN values properly. What's even more confusing
+is that sometimes depending on the order of operations, the gradient could vary (e.g. depending on how early
```
Comment (svekars) [inline prototype_source/maskedtensor_overview.rst:119]:
```suggestion
is that sometimes depending on the order of operations, the gradient could vary (for example, depending on how early
```

File: prototype_source/maskedtensor_overview.rst [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+    >>> mt
+    MaskedTensor(
+    [
+        [  0.4216,       --,       --],
+        [ -0.7676,       --,   1.4216],
+        [      --,       --,       --]
+    ]
+    )
+
+Suppose we want to calculate the softmax along `dim=0`. Note that the second column is "unsafe" (i.e. entirely
```
Comment (svekars) [inline prototype_source/maskedtensor_overview.rst:263]:
```suggestion
For example, we want to calculate the softmax along `dim=0`. Note that the second column is "unsafe" (i.e. entirely
```

File: prototype_source/maskedtensor_overview.rst [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+
+    >>> x = torch.empty(16).fill_(float('nan'))
+    >>> x
+    tensor([nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan])
+    >>> torch.nanmean(x)
+    tensor(nan)
+    >>> torch.mean(masked_tensor(x, ~torch.isnan(x)))
+    MaskedTensor(--, False)
+
+This is a similar problem to safe softmax where `0/0 = nan` when what we really want is an undefined value.
```
Comment (svekars) [inline prototype_source/maskedtensor_overview.rst:334]: Add Further Reading 

Further Reading
----------------
* Link to the next MaskedTensor Tutorial

File: prototype_source/maskedtensor_overview.rst [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+(an ambiguous return value), while MaskedTensor would more accurately indicate a masked out result.
+
+    >>> x = torch.empty(16).fill_(float('nan'))
+    >>> x
+    tensor([nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan])
+    >>> torch.nanmean(x)
+    tensor(nan)
+    >>> torch.mean(masked_tensor(x, ~torch.isnan(x)))
+    MaskedTensor(--, False)
+
```
Comment (svekars) [inline prototype_source/maskedtensor_overview.rst:333]: Add a Conclusion

Conclusion
------------

In this tutorial, we have learned about .....

**Review by jisaacso — APPROVED** [review]
LG, happy to rereview after rendering of code blocks
